Src/FluentAssertions/Common/MemberPath.cs

```diff
@@ -66,8 +74,33 @@ private bool IsChildOf(MemberPath candidate)
                    && candidateSegments.SequenceEqual(Segments.Take(candidateSegments.Length));
         }
 
+        /// <summary>
+        /// Determines whether the current path is the same as <paramref name="path"/> when ignoring any specific indexes.
+        /// </summary>
+        public bool IsEquivalentTo(string path)
+        {
+            return path.WithoutSpecificCollectionIndex() == dottedPath.WithoutSpecificCollectionIndex();
+        }
+
+        public bool HasSameParentAs(MemberPath path)
+        {
+            return ParentSegments.SequenceEqual(path.ParentSegments);
+        }
+
+        private string[] ParentSegments => Segments.Take(Segments.Length - 1).ToArray();
+
         private string[] Segments => segments ??= dottedPath.Split(new[] { '.', '[', ']' }, StringSplitOptions.RemoveEmptyEntries);
 
+        /// <summary>
+        /// Returns the name of the member the current path points to without its parent path.
+        /// </summary>
+        public string MemberName => Segments.Last();
+
+        /// <summary>
+        /// Gets a value indicating whether the current path contains an indexer like `[1]` instead of `[]`.
+        /// </summary>
+        public bool ContainsSpecificCollectionIndex => dottedPath.WithoutSpecificCollectionIndex() != dottedPath;
```

Comment:
Make this a method as it does a lot more than just returning a field

Reply:
It's `internal`, so I don't see much value in that.